csv/src/main/java/com/fasterxml/jackson/dataformat/csv/CsvParser.java

```diff
@@ -498,14 +498,13 @@ public void setCodec(ObjectCodec c) {
     }
 
     @Override
-    public void setSchema(FormatSchema schema)
+    public void setSchema(final FormatSchema schema)
     {
         if (schema instanceof CsvSchema) {
             _schema = (CsvSchema) schema;
-            String str = _schema.getNullValueString();
-            _nullValue = str;
+            _nullValue = _schema.getNullValueString();
         } else if (schema == null) {
-            schema = EMPTY_SCHEMA;
+            _schema = EMPTY_SCHEMA;
```

Comment:
this looks like a bug to me - the old code was updating the input param (`schema`) instead of the `_schema`.

Reply:
Yes, I concur this is a bug. I think I'll backport this in 2.17 as well.